jdaviz/configs/default/plugins/plot_options/plot_options.py

```diff
@@ -358,3 +369,110 @@ def vue_set_value(self, data):
+    def _update_stretch_histogram(self, msg={}):
+        if not self.stretch_function_sync.get('in_subscribed_states'):
+            # no (image) viewer with stretch function options
+            return
+        if not hasattr(self, 'viewer'):
+            # plugin hasn't been fully initialized yet
+            return
+        if not self.plugin_opened or not self.viewer.selected or not self.layer.selected:
+            # no need to make updates, updates will be redrawn when plugin is opened
+            # NOTE: this won't update when the plugin is shown but not open in the tray
+            return
+        if not isinstance(msg, dict) and not self.stretch_hist_zoom_limits:
+            # then this is from the limits callbacks and we don't want to waste resources
+            # IMPORTANT: this assumes the only non-observe callback to this method comes
+            # from state callbacks from zoom limits.
+            return
+
+        if self.multiselect and (len(self.viewer.selected) > 1 or len(self.layer.selected) > 1):
+            # currently only support single-layer/viewer.  For now we'll just clear and return.
+            # TODO: add support for multi-layer/viewer
+            bqplot_clear_figure(self.stretch_histogram)
+            return
+
+        viewer = self.viewer.selected_obj[0] if self.multiselect else self.viewer.selected_obj
+
+        # manage viewer zoom limit callbacks
+        if ((isinstance(msg, dict) and msg.get('name') == 'viewer_selected')
+                or not self.stretch_hist_zoom_limits):
+            vs = viewer.state
+            for attr in ('x_min', 'x_max', 'y_min', 'y_max'):
+                vs.add_callback(attr, self._update_stretch_histogram)
+        if isinstance(msg, dict) and msg.get('name') == 'viewer_selected':
+            viewer_label_old = msg.get('old')[0] if self.multiselect else msg.get('old')
+            if len(viewer_label_old):
+                vs_old = self.app.get_viewer(viewer_label_old).state
+                for attr in ('x_min', 'x_max', 'y_min', 'y_max'):
+                    vs_old.remove_callback(attr, self._update_stretch_histogram)
+
+        data = self.layer.selected_obj[0][0].layer if self.multiselect else self.layer.selected_obj[0].layer  # noqa
+        comp = data.get_component(data.main_components[0])
+
+        if self.stretch_hist_zoom_limits:
+            if hasattr(viewer, '_get_zoom_limits'):
+                # Viewer limits. This takes account of Imviz linking.
+                xy_limits = viewer._get_zoom_limits(data).astype(int)
+                x_limits = xy_limits[:, 0]
+                y_limits = xy_limits[:, 1]
+                x_min = x_limits.min()
+                x_max = x_limits.max()
+                y_min = y_limits.min()
+                y_max = y_limits.max()
+
+                # TODO: this doesn't seem to exactly match the full image when zoom is reset
+                sub_data = comp.data[y_min:y_max, x_min:x_max].ravel()

```

Comment:
@pllim - can you take another look at this logic (ported directly from #2097)?  It doesn't seem to be working in my example case - when the image is all the way zoomed out, I'd expect the same histogram both with and without the switch flipped, but that isn't the case.

Reply:
(fixed by ensuring the `x_min`/`y_min` are not set below zero even if the viewer limits are technically negative)